lib/migrator.js

```diff
@@ -48,27 +68,32 @@ internals.executeMigration = async (migration, rollback) => {
     catch (err) {
 
         // Even if our symlinking fails, we need to manually cleanup our temporary directory because
-        // the unsafeCleanup: true option we pass to tempDirAsync (Tmp.dir) causes the tool to not delete the temp directory
-        // if it has entries, which it will if we fail partway through
-        tmpResults.cleanup();
+        // the unsafeCleanup: true option we pass to tempDir (Tmp.dir promisified) causes Tmp to not delete the temp directory if it has entries,
+        // which it will if we fail partway through
+        tmpCleanup();
+        throw err;
     }
 
-    // TODO Review / fix tmpAssets.tmpPath
-    const relMigrationsDir = Path.relative(process.cwd(), tmpResults.tmpPath);
+    const relMigrationsDir = Path.relative(process.cwd(), tmpPath);
     const latestOrRollback = rollback ? 'rollback' : 'latest';
 
     await knex.migrate[latestOrRollback]({ directory: relMigrationsDir });
 
-    // TODO does this work? What happens when a callback takes 2 non-error parameters?
-    tmpResults.cleanup();
+    tmpCleanup();
+
 };
 
-internals.readMigrationDir = async (migrationDir, symlinkConfig) => {
+internals.readMigrationDir = async (migrationDir, symlinkStore) => {
```

Comment:
This function does read directories, but the important action it takes is really symlinking each file in a migration directory to a file in a temp directory.